streams/src/main/java/org/apache/kafka/streams/kstream/Serialized.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream;
+
+import org.apache.kafka.common.serialization.Serde;
+
+/**
+ * The class that is used to capture the key and value {@link Serde}s used when performing
+ * {@link KStream#groupBy(KeyValueMapper, Serialized)} and {@link KStream#groupByKey(Serialized)} operations.
+ *
+ * @param <K> the key type
+ * @param <V> the value type
+ */
+public class Serialized<K, V> {
+
+    private Serde<K> keySerde;
+    private Serde<V> valueSerde;
+
+    private Serialized(final Serde<K> keySerde,
+                       final Serde<V> valueSerde) {
+        this.keySerde = keySerde;
+        this.valueSerde = valueSerde;
+    }
+
+    public Serde<K> keySerde() {
```

Comment:
See my other comment about hiding these getters from users.

Reply:
I'm going to merge this as is. If we can think of a way to remove these that will work, then i'll file another JIRA to do this.

Reply:
Sounds good. Thanks @dguy